api/api-backups-v2/src/main/java/com/thoughtworks/go/apiv2/backups/BackupsControllerV2.java

```diff
@@ -0,0 +1,101 @@
+import com.thoughtworks.go.spark.Routes;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import com.thoughtworks.go.util.SystemEnvironment;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import spark.Request;
+import spark.Response;
+
+import java.io.IOException;
+
+import static com.thoughtworks.go.api.util.HaltApiResponses.haltBecauseConfirmHeaderMissing;
+import static com.thoughtworks.go.spark.Routes.Backups.ID_PATH;
+import static spark.Spark.*;
+
+@Component
+public class BackupsControllerV2 extends ApiController implements SparkSpringController {
+
+    private static final HeaderConstraint HEADER_CONSTRAINT = new HeaderConstraint(new SystemEnvironment());
+    private static final String RETRY_INTERVAL_IN_SECONDS = "30";
+
+    private final ApiAuthenticationHelper apiAuthenticationHelper;
+    private BackupService backupService;
+
+    @Autowired
+    public BackupsControllerV2(ApiAuthenticationHelper apiAuthenticationHelper, BackupService backupService) {
+        super(ApiVersion.v2);
+        this.apiAuthenticationHelper = apiAuthenticationHelper;
+        this.backupService = backupService;
+    }
+
+    @Override
+    public String controllerBasePath() {
+        return Routes.Backups.BASE;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("/*", mimeType, this::setContentType);
+
+            before("", mimeType, this::verifyConfirmHeader);
+
+            before("", this.mimeType, this.apiAuthenticationHelper::checkAdminUserAnd403);
+            before("/*", this.mimeType, this.apiAuthenticationHelper::checkAdminUserAnd403);
+
+            post("", mimeType, this::create);
+            get(ID_PATH, mimeType, this::show);
+            exception(RecordNotFoundException.class, this::notFound);
+        });
+    }
+
+    public String create(Request request, Response response) {
+        ServerBackup backup = backupService.scheduleBackup(currentUsername());
+        response.status(202);
+        response.header("Location", Routes.Backups.serverBackup(String.valueOf(backup.getId())));
+        response.header("Retry-After", RETRY_INTERVAL_IN_SECONDS);
+        return NOTHING;
+    }
+
+    public String show(Request request, Response response) throws IOException {
+        String backupId = request.params("id");
+        ServerBackup backup = backupService.getServerBackup(backupId);
+        if (null == backup) {
+            throw new RecordNotFoundException();
+        }
+        return writerForTopLevelObject(request, response, outputWriter -> BackupRepresenter.toJSON(outputWriter, backup));
+    }
+
+    private void verifyConfirmHeader(Request request, Response response) {
+        if (!HEADER_CONSTRAINT.isSatisfied(request.raw())) {
+            throw haltBecauseConfirmHeaderMissing();
```

Comment:
@marques-work I did modify `HeaderConstraint` to check for both legacy and new header and then used it. I did not see the `verifyContentType`, maybe i'll just switch to using it.

Reply:
Also I noticed that verifyContentType does not check if the x-gocd-confirm header value is set to true, only checks its existence unlike HeaderConstraint

Reply:
Also I noticed that the `verifyContentType` only checks for the presence of the confirm header and does not if its value is set to true unlike the `HeaderConstraint`

Reply:
@prateekbaheti that's true, I do see that it's just a presence check. While it's probably safe to add that check to `verifyContentType()` as I beleive that was the intent, I still would be hesitant to make that change since this is used everywhere practically. Maybe check with @ketan if he thinks it's sufficient to use `verifyContentType()` for a header check?

Reply:
@marques-work Yes agreed, changing that would affect quite a few places.